functions/New-DbaAvailabilityGroup.ps1

```diff
@@ -496,7 +496,11 @@ function New-DbaAvailabilityGroup {
                     # Add replicas
                     $null = Add-DbaAgReplica @replicaparams -EnableException -SqlInstance $second
                 } catch {
-                    Stop-Function -Message "Failure" -ErrorRecord $_ -Target $second -Continue
+                    $msg = $_.Exception.InnerException.InnerException.Message
```

Comment:
We have `Get-ErrorMessage` that tests for the proper inner exception and `Stop-Function` calls that command: https://github.com/sqlcollaborative/dbatools/blob/e0683a106353440e7b53d0dcaae7b7766f2e578c/internal/functions/flowcontrol/Stop-Function.ps1#L185

Is there a reason this is needed?

Reply:
No, probably not. I was just not aware of this internal function and just copied that from another catch block from within that command.
I just cleaned up the catch blocks and pushed a new commit. Can you please have a look?